x-pack/plugin/security/qa/jwt-realm/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRestIT.java

```diff
@@ -68,10 +74,112 @@
+        settings.put("xpack.security.authc.realms.jwt.jwt1.required_claims.token_use", "id");
+        settings.put("xpack.security.authc.realms.jwt.jwt1.required_claims.version", "2.0");
+        settings.put("xpack.security.authc.realms.jwt.jwt1.client_authentication.type", "NONE");
+        // Use default value (RS256) for signature algorithm
+        settings.put("xpack.security.authc.realms.jwt.jwt1.pkc_jwkset_path", "rsa.jwkset");
+
+        // Place native realm after JWT realm to verify realm chain fall-through
+        settings.put("xpack.security.authc.realms.native.lookup_native.order", "2");
+
+        settings.put("xpack.security.authc.realms.jwt.jwt2.order", "3");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.token_type", "access_token");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.fallback_claims.sub", "email");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.fallback_claims.aud", "scope");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.allowed_issuer", "my-issuer");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.allowed_subjects", serviceSubject.get());
+        settings.put("xpack.security.authc.realms.jwt.jwt2.allowed_audiences", "es01,es02,es03");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.allowed_signature_algorithms", "HS256,HS384");
+        // Both email or sub works because of fallback
+        if (randomBoolean()) {
+            settings.put("xpack.security.authc.realms.jwt.jwt2.claims.principal", "email");
+        } else {
+            settings.put("xpack.security.authc.realms.jwt.jwt2.claims.principal", "sub");
+        }
+        settings.put("xpack.security.authc.realms.jwt.jwt2.claim_patterns.principal", "^(.*)@[^.]*[.]example[.]com$");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.required_claims.token_use", "access");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.authorization_realms", "lookup_native");
+        settings.put("xpack.security.authc.realms.jwt.jwt2.client_authentication.type", "shared_secret");
+
+        // Place PKI realm after JWT realm to verify realm chain fall-through
+        settings.put("xpack.security.authc.realms.pki.pki_realm.order", "4");
+
+        settings.put("xpack.security.authc.realms.jwt.jwt3.order", "5");
+        if (explicitIdTokenType) {
+            settings.put("xpack.security.authc.realms.jwt.jwt3.token_type", "id_token");
+        }
+        settings.put("xpack.security.authc.realms.jwt.jwt3.allowed_issuer", "jwt3-issuer");
+        settings.put("xpack.security.authc.realms.jwt.jwt3.allowed_audiences", "[jwt3-audience]");
+        settings.put("xpack.security.authc.realms.jwt.jwt3.allowed_signature_algorithms", "[HS384, HS512]");
+        settings.put("xpack.security.authc.realms.jwt.jwt3.claims.principal", "sub");
+        settings.put("xpack.security.authc.realms.jwt.jwt3.client_authentication.type", "Shared_Secret");
```

Comment:
Nit: I know this is just copy and paste but noticed that this one is capitalized -- wondering if we should re-use VALID_SHARED_SECRET here instead -- at least when I change it, the tests still pass 🤷

Reply:
I don't quite follow your suggestion.
This is setting the type, not the value of the secret.

From memory, I think the `shared_secret` and `Shared_Secret` is trying to loosely test that case doesn't matter for that setting, so I don't think we want to change it.

Reply:
I'm going to merge, but happy to open a new PR if there's something that you think could be improved here.

Reply:
Ah crap, I overlooked that this was `.type` -- no changes needed and thanks for the clarification!